dat/watchlist.js

```diff
@@ -0,0 +1,111 @@
+const EventEmitter = require('events')
+const emitStream = require('emit-stream')
+
+// dat modules
+const archivesDb = require('../dbs/archives')
+const datLibrary = require('../dat/library')
+const watchlistDb = require('../dbs/watchlist')
+
+class BadParamError extends Error {
```

Comment:
There's not really any point in using the custom errors. Internally we're not differentiating by the error types.

Any time we need an error to bubble to userland websites, we have to use https://www.npmjs.com/package/beaker-error-constants because the webview-preload has to import the error definitions independently of the backend. (See https://github.com/pfrazee/pauls-electron-rpc/blob/master/lib/import-api.js#L231). As a result, if the custom error isnt defined in beaker-error-constants, it just emits as a generic Error in the frontend.

Reply:
Oh okay that makes sense. I just copied that over early on starting this when referencing history API, I believe. I can go ahead and remove these custom errors.